src/backend/utils/adt/agtype.c

```diff
@@ -3968,8 +4093,14 @@ Datum agtype_access_operator(PG_FUNCTION_ARGS)
         container = NULL;
     }
 
+    pfree(args);
+    pfree(types);
+    pfree(nulls);
+
     /* serialize and return the result */
-    return AGTYPE_P_GET_DATUM(agtype_value_to_agtype(container_value));
+    result = agtype_value_to_agtype(container_value);
+
```

Comment:
`container_value` should be freed here?

Reply:
Unfortunately, due to how objects and arrays are iterated through, it can't be freed.